docs/repository/databinding.md

```diff
@@ -0,0 +1,133 @@
+
+## Data binding using MapBinder
+Plugin comes with a [MapBinder][]{.new-tab} Which is used by a Repository to perform databinding.
+Plugin configures [EntityMapBinder][]{.new-tab} as default implementation of [MapBinder][]{.new-tab}. [EntityMapBinder][]{.new-tab} is similar 
+to grails data binder in the sense that it uses registered value converters and fallbacks to spring ConversionService.
+However entityMapBinder is optimized to convert most commonly encountered property types such as Numbers and Dates 
+without going through the converters, thus resulting in faster performance.
+
+**Example**
+
+```groovy
+class SomeService {
+    @Autowired
+    EntityMapBinder binder
+
+    void foo(Map params) {
+        Org org = new Org()
+        binder.bind(org, params)
+    }
+}
+
+```
+
+**Using custom MapBinder**  
+By default all Repositories use the default [EntityMapBinder][]{.new-tab} for databinding. However when a Repository is explicitly 
+created for a domain class, and if required, a custom MapBinder implementation can be used to perform databinding as per the need.
+
+```groovy
+
+class CustomMapBinder implements MapBinder {
+
+    public <T> GormEntity<T> bind(Map args, Object target, Map<String, Object> source, BindAction bindAction) {
+        //implement  
+    }
+
+    public <T> GormEntity<T> bind(Object target, Map<String, Object> source, BindAction bindAction) {
+        //implement
+    }
+
+}
+```
+then register the bean 
+
+```java
+beans = {
+    customMapBinder(CustomMapBinder) 
+}
+```
+
+```groovy
+class OrgRepo implements GormRepo<Org> {
+    
+    @Autowired
+    @Qualifier("customMapBinder")
+    CustomMapBinder mapBinder
+    
+    .........   
+}
+
+```
+
+This will make the OrgRepo use CustomMapBinder for data binding.
+
+
+## Binding Associations
+Gorm tools MapBinder handles associations little differently than Grails data binder for performance reasons.
+Associations are handled differently for create vs update.  
+
+- For create, if the association belongs to the domain which is being bound, new instance will be created. Else if the map contains id, reference to existing instance will be set.
```

Comment:
It’s not clear to me or here in the docs what the difference is in the rules between update and create

Reply:
.. are there benefits to having a difference?

Reply:
Difference between create and update is, for update new instances are never created, just set to existing if id is provided. Association is deep bound only if it belongs to original target. As per the description on ticket